<launch>
  <arg name="choreonoid_arg"    default=""/>
  <arg name="model_file"        default=""/>
  <arg name="model_namespace"   default=""/>
  <arg name="control_namespace" default=""/>
  <arg name="controllers"       default="joint_state_controller" />
  <arg name="load_robot_description" default="true" />

  <!-- choreonoid_ros -->
  <include file="$(find choreonoid_ros)/launch/choreonoid.launch">
    <arg name="choreonoid_arg" value="$(arg choreonoid_arg)"/>
  </include>

  <!-- ROS Control -->
  <group ns="$(arg control_namespace)">
    <!-- load the controllers -->
    <node name="robot_controller_spawner" pkg="controller_manager"
          type="spawner" output="screen" respawn="true"
          args="$(arg controllers)"/>
  </group>

  <!-- For ROS (rviz, etc.) -->
  <group ns="$(arg model_namespace)">
    <param if="$(arg load_robot_description)"
           name="robot_description" command="$(find xacro)/xacro '$(arg model_file)'"/>

    <!-- state publisher -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true">
      <remap from="joint_states" to="$(arg control_namespace)/joint_states" />
    </node>
  </group>

</launch>
